File: online_judges/kattis/DownTime.java
package com.company;
import java.util.*;
public class DownTime {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n,k;
        n = sc.nextInt();
        k = sc.nextInt();
        int[] v = new int[n];
        Queue<Integer> q = new LinkedList<Integer>();
        for(int i=0;i<n;i++) {
            v[i] = sc.nextInt();
        }
        Arrays.sort(v);

        int cnt=1;

        for(int i=0;i<n;i++) {

            while(q.size() > 0){
                int x = q.peek();
                if(v[i] - x >= 1000) q.remove();
                else break;
            }

            q.add(v[i]);
            int need = q.size() / k + (q.size()%k ==0 ? 0 : 1 );

            if(need > cnt) cnt++;
        }

        System.out.println(cnt);
    }
}

File: online_judges/kattis/downtime.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    //ifstream cin("i.txt");
    int n,k;

    cin >> n >> k;

    vector<int> v(n);
    for(int i=0;i<n;i++) {
        cin >> v[i];
    }

    sort(v.begin(), v.end());


    queue<int> q;
    int cnt=1;
    for(int i=0;i<n;i++) {
        while(!q.empty()){
            int x = q.front();
            if(v[i] - x >= 1000) q.pop();
            else break;
        }
        q.push(v[i]);
        int need = q.size() / k + (q.size()%k ==0 ? 0 : 1 );

        if(need > cnt) cnt ++;
    }

    cout << cnt << endl;
    return 0;
}
